actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -2,8 +2,14 @@ module ActionController
   module ImplicitRender
     def send_action(method, *args)
       ret = super
-      default_render unless performed?
-      ret
+      return ret if performed?
+
+      if template_exists?(method, _prefixes, variants: request.variant)
+        default_render
+      else
+        logger.info "No template found for #{self.class.name}#{action_name}, rendering head :no_content"
+        head :no_content
+      end
```

Comment:
Does it matter that we're changing the return value here? Presumably it's not causing any test failures... but the previous version was clearly going out of its way.